packages/storage/src/implementation/error.ts

```diff
@@ -53,8 +76,17 @@ export class FirebaseStorageError extends FirebaseError {
 
 export const errors = {};
 
-export type Code = string;
-export const Code = {
+/**
+ * @public
+ * Error codes that can be attached to `FirebaseStorageError`s.
+ */
+export type StorageErrorCode = string;
+
+/**
+ * @public
+ * Error codes that can be attached to `FirebaseStorageError`s.
+ */
+export const StorageErrorCode = {
```

Comment:
Doesn't have to be public, changed exported `FirebaseStorageError` to export the public type to avoid requiring this. I made it an enum but if it's a const enum I can't import it into other files, and it's used a constant in equality checks in task.ts for example.

Reply:
You can import const enum in other files in the same package and use it as values. it's that you can't import it from another package. The equality check should also work with const enum.

Reply:
I was getting build errors and finally figured it out, you can export const enums but not from the same file as code? Functions? I'm not sure why. So I moved it to `constants.ts` and it works now. Is that the right solution?

Reply:
Weird. I moved it back to error.ts, and I didn't get any build error.

Reply:
Works in its original place after clean build, restored.